pkg/instance/service.go

```diff
@@ -124,6 +155,8 @@ func (s service) validateNoCycles(instances []*model.DeploymentInstance) (graph.
 					return nil, fmt.Errorf("instance %q requires %q more than once", src.Name, requiredStack.Name)
 				} else if errors.Is(err, graph.ErrEdgeCreatesCycle) {
 					return nil, fmt.Errorf("link from instance %q to stack %q creates a cycle", src.Name, requiredStack.Name)
+				} else if errors.Is(err, graph.ErrVertexNotFound) {
+					return nil, fmt.Errorf("link from %q to stack %q required", requiredStack.Name, src.StackName)
```

Comment:
You are calling `validateNoCycles` from 4 methods. Does this error message make sense for all these cases? 🤔 `graph.ErrVertexNotFound` generally means that the required stack (vertex) is missing. The link (edge) is actually present but only one of its vertices is.

1. `SaveInstance(instance *model.DeploymentInstance)` - will this error be returned if I am saving a `core` without having saved a `db` first? Or will that be caught in https://github.com/dhis2-sre/im-manager/blob/a546c81c142bc767b26b872d391e88a1d959388d/pkg/instance/service.go#L228 `resolveConsumedParameters`
2. `DeployDeployment(token string, deployment *model.Deployment)`
3. `DeleteDeployment(deployment *model.Deployment)`
4. `DeleteInstance(deploymentId, instanceId uint)`